client/src/utils/fight/block.ts

```diff
@@ -4,30 +4,40 @@ import { Application } from 'pixi.js';
 
 import { sound } from '@pixi/sound';
 import findFighter, { AnimationFighter } from './utils/findFighter';
+import { knockBack } from './utils/knockBack';
+import { playDustEffect } from './utils/playVFX';
 
 const block = async (
   app: Application,
   fighters: AnimationFighter[],
   step: BlockStep,
+  speed: React.MutableRefObject<number>,
 ) => {
   if (!app.loader) {
     return;
   }
+
   const fighter = findFighter(fighters, step.f);
   if (!fighter) {
     throw new Error('Fighter not found');
   }
 
-  const animationEnded = fighter.animation.waitForEvent('block:end');
-
+  const animationsEnded = [
+    fighter.animation.waitForEvent('block:end'),
+    // Slight knockBack
+    knockBack(fighter, speed, 9, 0.25),
```

Comment:
Here, the knockback animation starts before changing the animation to block, it should be the opposite

Reply:
It is the solution that looks the best on testing.

Also, the block animation starts on frame 3 : on('block', 3);

They are actually played together.

Reply:
Oh ok I got your point, i'll change that